File: design-patterns/go/interpreter/logical_operation/xor_operation.go
// xor_operation.go

package main

type XorOperation struct {
	op1 Operation
	op2 Operation
}

func NewXorOperation(op1 Operation, op2 Operation) (xorOperation *XorOperation) {
	xorOperation = &XorOperation{}
	xorOperation.op1 = op1
	xorOperation.op2 = op2
	return
}

func (xorOperation *XorOperation) Execute(opContext string) bool {
	return xorOperation.op1.Execute(opContext) != xorOperation.op2.Execute(opContext)
}

File: design-patterns/ts/interpreter/logical-operation/xor-operation.ts
// xor-operation.ts

import Operation from "./operation";

class XorOperation implements Operation {
    private op1: Operation;
    private op2: Operation;

    constructor(op1: Operation, op2: Operation) {
        this.op1 = op1;
        this.op2 = op2;
    }

    execute(opContext: string): boolean {
        return this.op1.execute(opContext) != this.op2.execute(opContext);
    }
}

export default XorOperation;